src/plugins/intel_npu/src/common/src/igraph.cpp

```diff
@@ -13,6 +13,38 @@ constexpr std::size_t DEFAULT_BATCH_SIZE = 1;
 
 namespace intel_npu {
 
+IONodeMetadata::IONodeMetadata(const ArgumentDescriptor& d) : descriptor(d) {}
+
+std::optional<size_t> extract_batch_impl(const ov::Shape& shape,
+                                         std::optional<ov::PartialShape> partial_shape_ir,
+                                         std::optional<ov::PartialShape> partial_shape_compiler,
+                                         size_t index) {
+    if (partial_shape_ir.has_value() && partial_shape_compiler.has_value()) {
+        if (partial_shape_ir.value()[index].is_dynamic() &&
```

Comment:
I've reviewed the code again and I have to admit, that I'm starting to dislike this new check: 
1) It reveals the details of the plugin-batch implementation specifics (fromCompiler != fromIR)
2) it will become wrong  when we have shifted to compiler-batch paradigm, as in this case we may have fromCompiler == fromIR

Reply:
To be honest, the previous solution seems less risky to me as well.